processing/src/main/java/io/druid/query/groupby/epinephelinae/GroupByMergingQueryRunnerV2.java

```diff
@@ -154,6 +160,22 @@ public GroupByMergingQueryRunnerV2(
     final boolean hasTimeout = QueryContexts.hasTimeout(query);
     final long timeoutAt = System.currentTimeMillis() + queryTimeout;
 
+    final Supplier<ResourceHolder<ByteBuffer>> combineBufferSupplier = new Supplier<ResourceHolder<ByteBuffer>>()
+    {
+      private boolean initialized;
+      private ResourceHolder<ByteBuffer> buffer;
+
+      @Override
+      public ResourceHolder<ByteBuffer> get()
+      {
+        if (!initialized) {
+          buffer = processingBufferPool.take();
```

Comment:
this changes an assumption made at historicals and realtime task processes that processing buffer pool would only be asked for buffers from processing thread (one per thread) ... that would keep the usage of processing buffer pool restricted and no processing thread would need to block to get a buffer.
( came across this while doing https://github.com/druid-io/druid/pull/5345 )

I think that this buffer should really come from merge pool. what do you think? @gianm @jihoonson

Reply:
It should definitely not be from the processing pool. The merge pool might be the right place, however if so, this should be combined with the other code in the same file that pulls from the merge pool in order to make it atomic. Otherwise we could get deadlocks.

Reply:
@jihoonson what do you think?

Reply:
@jihoonson  please feel free to do the patch. thanks.

Reply:
@jihoonson nvmd , let me fix it in https://github.com/druid-io/druid/pull/5345 . its a small change.

Reply:
@himanshug thanks. I'll review that PR.